### PR TITLE
Deprecate Source.fromRawBytes

### PR DESCRIPTION
**Pull request by som-snytt**
Follow-up https://github.com/scala/bug/issues/6407

### CODE REVIEW

File: src/library/scala/io/Source.scala [hunk 1]
```diff
@@ -115,8 +115,9 @@ object Source {
   /** Create a `Source` from array of bytes, assuming
    *  one byte per character (ISO-8859-1 encoding.)
    */
+  @deprecated("Provide an explicit encoding", since="2.13.9")
```
Comment (SethTisue) [inline src/library/scala/io/Source.scala]: "Use `rawBytes` and provide an explicit encoding"?

Comment (som-snytt) [reply src/library/scala/io/Source.scala]: `fromBytes`, but anyone using `Source` will know that from the many hours they have wasted over the years with this API.

Comment (som-snytt) [reply src/library/scala/io/Source.scala]: `deprecated("Did you even read the Scaladoc")`

Comment (SethTisue) [reply src/library/scala/io/Source.scala]: oh, `fromBytes`, right 😬

Comment (SethTisue) [reply src/library/scala/io/Source.scala]: I force-pushed an adjusted wording.

**Review by SethTisue — APPROVED** [review]

**Pull request merged**